elastic/integrations
# [azure_network_watcher_nsg] Initial release of the azure network watcher nsg

## Proposed commit message
Create New integration package azure_network_watcher_nsg.
- Added data stream.
- Added data collection logic for log data stream.
- Added the ingest pipeline for log data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for log data stream.
- Added system test cases for log data stream.


## Checklist



## How to test this PR locally



- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/azure_network_watcher_nsg directory.
- Run the following command to run tests. 
> elastic-package test 

```
--- Test results for package: azure_network_watcher_nsg - START ---
╭───────────────────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE                   │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├───────────────────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ azure_network_watcher_nsg │ log         │ static    │ Verify sample_event.json │ PASS   │ 108.600754ms │
╰───────────────────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_nsg - END   ---
Done
--- Test results for package: azure_network_watcher_nsg - START ---
╭───────────────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE                   │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ azure_network_watcher_nsg │ log         │ system    │ default   │ PASS   │ 35.829069155s │
╰───────────────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: azure_network_watcher_nsg - END   ---
Done
--- Test results for package: azure_network_watcher_nsg - START ---
╭───────────────────────────┬─────────────┬───────────┬────────────────────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE                   │ DATA STREAM │ TEST TYPE │ TEST NAME                                                                          │ RESULT │ TIME ELAPSED │
├───────────────────────────┼─────────────┼───────────┼────────────────────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ azure_network_watcher_nsg │             │ asset     │ dashboard azure_network_watcher_nsg-0c8f22c3-153c-4466-a97c-2f250fbda5de is loaded │ PASS   │        824ns │
│ azure_network_watcher_nsg │             │ asset     │ search azure_network_watcher_nsg-5284cd1a-dab3-4b98-867e-02ed2005c4fe is loaded    │ PASS   │        150ns │
│ azure_network_watcher_nsg │             │ asset     │ search azure_network_watcher_nsg-c63b3237-5106-44b8-9ac9-52c7a1c4feeb is loaded    │ PASS   │        152ns │
│ azure_network_watcher_nsg │             │ asset     │ search azure_network_watcher_nsg-cfae4d90-dc5e-48c5-8ca4-b0755e169986 is loaded    │ PASS   │        109ns │
│ azure_network_watcher_nsg │ log         │ asset     │ index_template logs-azure_network_watcher_nsg.log is loaded                        │ PASS   │        216ns │
│ azure_network_watcher_nsg │ log         │ asset     │ ingest_pipeline logs-azure_network_watcher_nsg.log-0.1.0 is loaded                 │ PASS   │        137ns │
╰───────────────────────────┴─────────────┴───────────┴────────────────────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_nsg - END   ---
Done
--- Test results for package: azure_network_watcher_nsg - START ---
╭───────────────────────────┬─────────────┬───────────┬──────────────┬────────┬──────────────╮
│ PACKAGE                   │ DATA STREAM │ TEST TYPE │ TEST NAME    │ RESULT │ TIME ELAPSED │
├───────────────────────────┼─────────────┼───────────┼──────────────┼────────┼──────────────┤
│ azure_network_watcher_nsg │ log         │ pipeline  │ test-nsg.log │ PASS   │   7.320561ms │
╰───────────────────────────┴─────────────┴───────────┴──────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_nsg - END   ---
Done
```
## Related issues

- Closes #8941 


## Screenshot
![image](https://github.com/elastic/integrations/assets/123942854/5ca728cc-3b5f-4db9-9a91-2b458ec1cf1b)
![image](https://github.com/elastic/integrations/assets/123942854/f6771f17-98eb-4f8f-9fca-821fb16adb59)
